ewatercycle/models/marrmot.py

```diff
@@ -42,42 +42,70 @@ def _generate_work_dir(work_dir: PathLike = None):
+
+    Args:
+        version: pick a version for which an ewatercycle grpc4bmi docker image is available.
+        forcing: a forcing file or a forcing data object. See format forcing file in `model implementation <https://github.com/wknoben/MARRMoT/blob/8f7e80979c2bef941c50f2fb19ce4998e7b273b0/BMI/lib/marrmotBMI_oct.m#L15-L19>`_.
+            If forcing file contains parameter and other settings, those are used and can be changed in :py:meth:`steup`.
 
     Attributes:
         bmi (Bmi): Basic Modeling Interface object
-        work_dir (PathLike): Working directory for the model where it can read/write files
 
     Example:
         See examples/marrmotM01.ipynb in `ewatercycle repository <https://github.com/eWaterCycle/ewatercycle>`_
     """
     model_name = "m_01_collie1_1p_1s"
-    """Name of model in Matlab code"""
+    """Name of model in Matlab code."""
+    available_versions = ["2020.11"]
+    """Versions for which ewatercycle grpc4bmi docker images are available."""
 
-    def __init__(self):
+    def __init__(self, version: str, forcing: PathLike):
+        """Construct MarrmotM01 with initial values. """
         super().__init__()
+        self.version = version
+        if CFG['container_engine'].lower() == 'singularity':
+            self._set_singularity_image()
+        elif CFG['container_engine'].lower() == 'docker':
+            self._set_docker_image()
+        else:
+            raise ValueError(
+                f"Unknown container technology in CFG: {CFG['container_engine']}"
+            )
         self._parameters = [1000.0]
         self.store_ini = [900.0]
         self.solver = Solver()
-        self.forcing_file: PathLike = None
+
+        self.forcing = forcing
+        self._check_forcing(self.forcing)
+
+    def _set_docker_image(self):
+        images = {
+            '2020.11': 'ewatercycle/marrmot-grpc4bmi:2020.11'
+        }
+        self.docker_image = images[self.version]
+
+    def _set_singularity_image(self):
+        images = {
+            '2020.11': 'ewatercycle-marrmot-grpc4bmi_2020.11.sif'
+        }
+        self.singularity_image = CFG['singularity_dir'] / images[self.version]
 
     # unable to subclass with more specialized arguments so ignore type
     def setup(self,  # type: ignore
-              forcing: Union[ForcingData, PathLike],
-              maximum_soil_moisture_storage: float = 1000.0,
-              initial_soil_moisture_storage: float = 900.0,
+              maximum_soil_moisture_storage: float = None,
+              initial_soil_moisture_storage: float = None,
               start_time: datetime = None,
               end_time: datetime = None,
```

Comment:
Start time and end time are not listed under `model.parameters`. I would expect the output of that property to match the input arguments for setup.

Reply:
It would also be helpful if they can be passed as strings instead of datetime objects. So if `model.parameters` provides  `(start_time, "20210101")`, then `model.setup(start_time="20210102")` might be more convenient than `start_time = datetime.Datetime(2020, 1, 1))`

Reply:
We could use `self.forcing_start_time` for the default `start_time`. 

For datetime as string I would like to standardize on [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) format.
Also the forcing uses UTC while dates given setup could be timezone less, which should be handled elegantly.

Reply:
now, the time of model in UTC and ISO format string can be given as 'YYYY-MM-DDTHH:MM:SS+00:00'. and start/end time are added to mode.parameters.

Reply:
What happens when you do `model.setup(start_time="2021-01-15T00:00:00+05:00")` will mat file be set to 2021-01-15 or 2021-01-16? Can a test be added

Reply:
The last part i.e. `+00:00` is a shift in `HH:MM` toward the UTC time. In your example, the shift is 5 hours and it returns `datetime.datetime(2021, 1, 15, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=18000)))`, and it is still 2021, 1, 15, 0, 0. 
I think we should consider the UTC time as a standard (like the time in forcing data) and add a function to check that `start_time` is either in this format 'YYYY-MM-DDTHH:MM:SS+00:00' or this 'YYYY-MM-DDTHH:MM:SS' , and also to inform the user that the given `start_time` should be a UTC time.

Reply:
Agree, instead of looking at string if timezone == UTC we could use datetime.tzinfo

```
>>> from datetime import datetime, timezone, timedelta
>>> datetime(2021, 1, 15, 0, 0, tzinfo=timezone(timedelta(seconds=18000))).tzinfo == timezone.utc
False
>>> datetime.now(timezone.utc).tzinfo == timezone.utc
True
>>> datetime.now().tzinfo == timezone.utc
False
```

Reply:
I changed the [datetime iso format] to a [datetime64 iso format] i.e. 'YYYY-MM-DDTHH:MM:SSZ' that is used by [numpy](https://numpy.org/doc/stable/reference/generated/numpy.datetime_as_string.html), and xarray. I also added a util function to parse and check if time is UTC. @sverhoeven please let me know if you are happy with the changes, then we merge the PR. Thanks.